policyengine_api/endpoints/policy.py

```diff
@@ -198,45 +206,58 @@ def get_policy_search(country_id: str) -> list:
-        "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
-        (country_id, f"%{query}%"),
-    )
-    if results is None:
-        return dict(
-            status="error",
-            message=f"Policy not found in {country_id}",
+    try:
+        results = database.query(
+            "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
+            (country_id, f"%{query}%"),
         )
-    else:
+
         results = results.fetchall()
 
-    # If unique_only is true, filter results to only include
-    # items where everything except ID is unique
-    if unique_only:
-        processed_vals = set()
-        new_results = []
-
-        # Compare every label and hash to what's contained in processed_vals
-        # If a label-hash set aren't already in processed_vals,
-        # add them to new_results
-        for policy in results[:]:
-            comparison_vals = policy["label"], policy["policy_hash"]
-            if comparison_vals not in processed_vals:
-                new_results.append(policy)
-                processed_vals.add(comparison_vals)
-
-        # Overwrite results with new_results
-        results = new_results
-
-    # Format into: [{ id: 1, label: "My policy" }, ...]
-    policies = [
-        dict(id=result["id"], label=result["label"]) for result in results
-    ]
-    return dict(
-        status="ok",
-        message=None,
-        result=policies,
-    )
+        if not results:
+            body = dict(
+                status="error",
+                message=f"No policies found for country {country_id} for query '{query}",
+            )
+            return Response(
+                json.dumps(body), status=404, mimetype="application/json"
+            )
+
+        # If unique_only is true, filter results to only include
+        # items where everything except ID is unique
+        if unique_only:
+            processed_vals = set()
+            new_results = []
+
+            # Compare every label and hash to what's contained in processed_vals
+            # If a label-hash set aren't already in processed_vals,
+            # add them to new_results
+            for policy in results[:]:
+                comparison_vals = policy["label"], policy["policy_hash"]
+                if comparison_vals not in processed_vals:
+                    new_results.append(policy)
+                    processed_vals.add(comparison_vals)
+
+            # Overwrite results with new_results
+            results = new_results
+
+        # Format into: [{ id: 1, label: "My policy" }, ...]
+        policies = [
+            dict(id=result["id"], label=result["label"]) for result in results
+        ]
+        body = dict(
+            status="ok",
+            message="Policies found",
+            result=policies,
+        )
+        return Response(
+            json.dumps(body), status=200, mimetype="application/json"
+        )
+    except:
```

Comment:
```suggestion
    except Exception as e:
```
One minor thing - we can actually return the error message with the response, so I think that'd be for the best.